QEP-label-connectors.rst

```diff
@@ -0,0 +1,109 @@
+.. _qep#[.#]:
+
+========================================================================
+QGIS Enhancement ??: Label connectors
+========================================================================
+
+:Date: October 2015
+:Author: Hugo Mercier
+:Contact: hugo dot mercier at oslandia dot com
+:Last Edited: 
+:Status:  
+:Version: QGIS 2.14
+
+Summary
+-------
+
+This proposal is aiming at adding to the labeling display engine the possibility for the user to define
+"connectors" between labels and the features they label.
+
+Current state
+-------------
+
+The current (2.11) version of QGIS allows to manually place labels when the output of the automatic placement mechanism is not enough.
+
+It relies on "data defined" properties. Indeed, the X and Y positions of a label as well as its rotation angle can be defined by a data column or any
+expression.
+
+Some graphical controls allow then to manually manipulate these properties. The state of a label can be changed from being automatically placed to being
+manually placed (pinning/unpinning). When they are pinned, labels can then be moved or rotated by interaction of the user. Positions and rotations
+data are then updated in columns of the layer.
+
+Proposed improvements
+---------------------
+
+It may be desirable to have visual "connectors" between the label and the feature it refers to. They are a generalization of "arrows", as it
+can be found in drawing programs: single arrows, straight lines or curves. Connectors are "attached" to a label: if one of the two endings
+(either the label or the feature) is moved, the shape of the connector is recomputed and a display refresh is needed.
+
+This QEP is not meant to give an exhaustive list of possible connectors. The idea is to have a generic implementation with the most common
+types of connectors and everything needed in the code to be able to add new types in the future.
+
+Possible types of connectors:
+- a straight line
+- two connected straight lines, with an horizontal part and a free part
+- three connected straight lines, with two horizontal parts and a vertical part
+- a curve with one bending point
+- a curve with two bending points
+
+.. image:: connector_types.jpg
+
+To actually draw the connectors, additional properties (aside X and Y position) are needed like:
+- style of the line (plain, dotted, dashed, etc.)
+- thickness of the line
+- color of the line
+- presence of arrow heads on both ends
+- style of the arrow heads
+- size of the arrow heads (absolute or relative)
+- color of the arrow heads
+
+Placement of the two ends also need additional properties:
+- position of the ending on the side of the label : up, down, right, left, center, etc.
```

Comment:
I also think an exact relative x/y placement should be allowed for the start and end. MapInfo doesn't allow this and only gives positions eg "left of label", "right of label", etc, and it's no-way near flexible enough for precise cartographic placement of leading lines.